src/Lib.P2P/Routing/DhtMessages.cs

```diff
@@ -0,0 +1,248 @@
+        /// ("try" here is loose, but this should signal "made strong effort, failed")
+        /// </summary>
+        CannotConnect = 3
+    }
+
+    /// <summary>
+    ///   Information about a peer.
+    /// </summary>
+    [ProtoContract]
+    public class DhtPeerMessage
+    {
+        /// <summary>
+        /// ID of a given peer. 
+        /// </summary>
+        /// <value>
+        ///   The <see cref="MultiHash"/> as a byte array,
+        /// </value>
+        [ProtoMember(1)]
+        public byte[] Id { get; set; }
+
+        /// <summary>
+        /// Addresses for a given peer
+        /// </summary>
+        /// <value>
+        ///   A sequence of <see cref="MultiAddress"/> as a byte array.
+        /// </value>
+        [ProtoMember(2)]
+        public byte[][] Addresses { get; set; }
+
+        /// <summary>
+        /// used to signal the sender's connection capabilities to the peer
+        /// </summary>
+        [ProtoMember(3)]
+        public ConnectionType Connection { get; set; }
+
+        /// <summary>
+        ///   Convert the message into a <see cref="Peer"/>.
+        /// </summary>
+        /// <param name="peer"></param>
+        /// <returns></returns>
+        public bool TryToPeer(out Peer peer)
+        {
+            peer = null;
+
+            // Sanity checks.
+            if (Id == null || Id.Length == 0)
+                return false;
+
+            var id = new MultiHash(Id);
+            peer = new Peer
+            {
+                Id = id
+            };
+            if (Addresses != null)
+            {
+                var x = new MultiAddress($"/ipfs/{id}");
+                peer.Addresses = Addresses
+                   .Select(bytes =>
+                    {
+                        try
+                        {
+                            var ma = new MultiAddress(bytes);
+                            ma.Protocols.AddRange(x.Protocols);
+                            return ma;
+                        }
+                        catch
+                        {
+                            return null;
+                        }
+                    })
+                   .Where(a => a != null)
+                   .ToArray();
+            }
+
+            return true;
+        }
+    }
+
+    /// <summary>
+    ///   The DHT message exchanged between peers.
+    /// </summary>
+    [ProtoContract]
+    public class DhtMessage
+    {
+        /// <summary>
+        /// What type of message it is.
+        /// </summary>
+        [ProtoMember(1)]
+        public MessageType Type { get; set; }
+
+        /// <summary>
+        ///   Coral cluster level.
+        /// </summary>
+        [ProtoMember(10)]
+        public int ClusterLevelRaw { get; set; }
+
+        /// <summary>
+        ///   TODO
+        /// </summary>
+        [ProtoMember(2)]
+        public byte[] Key { get; set; }
+
+        /// <summary>
+        ///   TODO
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)